### PR TITLE
Incorporate chainlib-bitcoin into bitcoind.js

### PR DESCRIPTION
**Pull request by pnagurny**

### CODE REVIEW

**Comment by braydonf** [comment]
- Submitted a PR that exposes `verifyScript` from bitcoind daemon since it's already available there: https://github.com/pnagurny/bitcoind.js/pull/1
- The example in the README needs to be updated and we can expose the main API methods on Node, similar to: https://github.com/bitpay/chainlib-bitcoin#api-documentation
- Wallet test data isn't needed and the `sendFunds` API method (we'll can add a broadcastTransaction method later)

**Comment by braydonf** [comment]
Also the integration tests and benchmarks need to be updated with the new path the daemon.

File: lib/block.js [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+  BaseBlock.call(this, obj);
+
+  this.bits = obj.bits;
+  this.nonce = obj.nonce || 0;
+
+}
+
+util.inherits(Block, BaseBlock);
+
+Block.prototype.validate = function(chain, callback) {
```
Comment (braydonf) [inline lib/block.js]: Do we need to validate the block, since the daemon will do this?

File: lib/db.js [hunk 2]
```diff
@@ -0,0 +1,381 @@
...
+    }
+  });
+
+  stream.on('close', function() {
+    if (error) {
+      return callback(error);
+    }
+
+    /*if(queryMempool) {
+      var mempoolOutputs = self._getMempoolOutputs(address);
```
Comment (braydonf) [inline lib/db.js]: `daemon.getMempoolOutputs()` is ready to be added in PR: https://github.com/bitpay/bitcoind.js/pull/33

File: lib/chain.js [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+  return bits;
+};
+
+Chain.prototype.getDifficultyFromBits = function getDifficultyFromBits(bits) {
+  var currentTarget = this.getTargetFromBits(bits);
+  var genesisTarget = this.getTargetFromBits(this.genesis.bits);
+  return genesisTarget.div(currentTarget);
+};
+
+Chain.prototype.getBlockWeight = function getBlockWeight(blockHash, callback) {
```
Comment (braydonf) [inline lib/chain.js]: A method to get this directly from Bitcoin Core is ready in PR: https://github.com/bitpay/bitcoind.js/pull/32

File: lib/db.js [hunk 4]
```diff
@@ -0,0 +1,384 @@
...
+      return callback(err);
+    }
+    callback(null, self.Block.fromBuffer(blockData));
+  });
+};
+
+DB.prototype.putBlock = function(block, callback) {
+  // block is already stored in bitcoind, but we need to update
+  // our prevhash index still
+  this._updatePrevHashIndex(block, callback);
```
Comment (braydonf) [inline lib/db.js]: We may be able to expose a method from the daemon, so that the prevHash index isn't need also (perhaps in another PR?).

**Comment by braydonf** [comment]
LGTM integration tests pass

File: lib/db.js (hunk 4 above)
Comment (pnagurny) [inline lib/db.js]: Yeah that would be great. Let's do it in another PR.

File: lib/block.js (hunk 1 above)
Comment (braydonf) [inline lib/block.js]: Spawned an issue: https://github.com/bitpay/bitcoind.js/issues/39

File: lib/db.js (hunk 4 above)
Comment (braydonf) [inline lib/db.js]: Spawned an issue: https://github.com/bitpay/bitcoind.js/issues/39

File: README.md [hunk 5]
```diff
@@ -1,25 +1,87 @@
...
+```js
+
+var BitcoinNode = require('bitcoind.js');
+
+var configuration = {
   directory: '~/.bitcoin',
   testnet: true
+};
+
+var node = new BitcoinNode(configuration);
```
Comment (braydonf) [inline README.md]: We'll need to revisit this, and make sure the configuration options are passed through correctly. Spawned issue: https://github.com/bitpay/bitcoind.js/issues/40

File: README.md [hunk 6]
```diff
@@ -1,25 +1,87 @@
...
-    // bitcoind is stopped
-  });
+Get Block
+
+```js
+var blockHash = '00000000d17332a156a807b25bc5a2e041d2c730628ceb77e75841056082a2c2';
+node.getBlock(blockHash, function(err, block) {
+  //...
 });
+```
```
Comment (braydonf) [inline README.md]: I think there are extra backticks here that are causing rending issues.

File: src/bitcoindjs.cc [hunk 7]
```diff
@@ -137,6 +137,43 @@ struct async_tx_data {
   Eternal<Function> callback;
 };
 
+/**
+ * Verify Scripts
+ */
+NAN_METHOD(VerifyScript) {
```
Comment (braydonf) [inline src/bitcoindjs.cc]: Spawned issue: https://github.com/bitpay/bitcoind.js/issues/41
